app/react/Library/components/SortButtons.js

```diff
@@ -153,41 +154,53 @@ export class SortButtons extends Component {
     return _search;
   }
 
+  renderDropdown(search, additionalSorts, { includeEvents = true } = {}) {
+    const { active } = this.state;
+    return (
+      <div
+        className={`Dropdown ${!includeEvents ? 'width-placeholder' : ''} order-by ${
+          active && includeEvents ? 'is-active' : ''
+        }`}
+      >
+        <ul className="Dropdown-list" onClick={includeEvents ? this.toggle : () => {}}>
+          {this.createSortItem('title', 'title', 'System', 'Title', {
+            isActive: search.sort === 'title',
+            search,
+            type: 'string',
+          })}
+          {this.createSortItem('creationDate', 'creationDate', 'System', 'Date added', {
+            isActive: search.sort === 'creationDate',
+            search,
+            type: 'date',
+          })}
+          {search.searchTerm && (
+            <li
+              key="relevance"
+              className={`Dropdown-option ${search.sort === '_score' ? 'is-active' : ''}`}
+            >
+              <a
+                className={`Dropdown-option__item ${search.sort === '_score' ? 'is-active' : ''}`}
+                onClick={() => (includeEvents ? this.handleClick('_score') : null)}
+              >
+                <span>{t('System', 'Search relevance')}</span>
+              </a>
+            </li>
+          )}
+          {additionalSorts}
+        </ul>
+      </div>
+    );
+  }
+
   render() {
     const { templates } = this.props;
     const search = this.validateSearch();
     const order = search.order === 'asc' ? 'up' : 'down';
     const additionalSorts = this.getAdditionalSorts(templates, search, order);
     return (
       <div className="sort-buttons">
-        <div className={`Dropdown order-by ${this.state.active ? 'is-active' : ''}`}>
-          <ul className="Dropdown-list" onClick={this.toggle.bind(this)}>
-            {this.createSortItem('title', 'title', 'System', 'Title', {
-              isActive: search.sort === 'title',
-              search,
-              type: 'string',
-            })}
-            {this.createSortItem('creationDate', 'creationDate', 'System', 'Date added', {
-              isActive: search.sort === 'creationDate',
-              search,
-              type: 'date',
-            })}
-            {search.searchTerm && (
-              <li
-                key="relevance"
-                className={`Dropdown-option ${search.sort === '_score' ? 'is-active' : ''}`}
-              >
-                <a
-                  className={`Dropdown-option__item ${search.sort === '_score' ? 'is-active' : ''}`}
-                  onClick={() => this.handleClick('_score')}
-                >
-                  <span>{t('System', 'Search relevance')}</span>
-                </a>
-              </li>
-            )}
-            {additionalSorts}
-          </ul>
-        </div>
+        {this.renderDropdown(search, additionalSorts, { includeEvents: false })}
```

Comment:
Was duplicating the dropdown as a placeholder the best strategy? I can be ignoring something, but couldn't this be fixed on the HTML/CSS?

Reply:
It’s a quick patch, because the approach to position the “| select all” button to the right of an absolutely positioned element is tricky.  More so because it’s size varies according to content.

Obviously there are better approaches, and even this one could just include the text of the select instead of all the structure. But the “reality” of this implementation is that it is an invisible text behind the floating text (which is configured to render identically to the floating element).

I do agree we need to rethink all that bar.

If you have any different approach for this scenario, I’m all ears!

Reply:
Let me elaborate a bit further: using an in-line element that servers as a placeholder for the floating (or absolutely positioned) element is perfectly valid and common approach. The fact that it holds the text and classes of the floating element to match its size is also the correct approach.

Where I agree is that there is no need to duplicate, hidden, the other <li> elements that are not the 'selected' one.  So, I can invest time in taking them away, but the duplication of the underlying sort button is, in my opinion, the correct approach.

It will take an extra effort to remove the 'non essential' <li> elements.  This can be done, for sure.

Reply:
Agreed on going with this implementation.